feat(issue-priority): Replace Priority search with Trends in the search backend

snigdhas opened the pull request:
Part 1/3 (4?) for https://github.com/getsentry/sentry/issues/65208:

Rename the Priority sort option to Trends to avoid confusing it with the new `Group.priority` column and feature. This PR changes the backend option to use `trends`. The frontend will continue to send `priority`, which we convert to `trends` until this rename is complete. 

Some tests have been duplicated to confirm we still handle `priority` as expected for backwards compatibility. 

Up next: we will need to migration `priority` to `trends` in `SavedSearch.sort`.

snigdhas committed:
Rename classes to use Trends

snigdhas committed:
Replace priority with trends in the search backend

snigdhas committed:
Replace priority with trends in the search backend

snigdhas committed:
Fix tests

snigdhas committed:
Fix tests

snigdhas committed:
More renames

snigdhas committed:
Add test for trend sort

snigdhas committed:
Add test for trend sort

snigdhas committed:
Add back priority to supported options

snigdhas committed:
Revert label change

snigdhas commented on src/sentry/models/savedsearch.py:
```diff
@@ -27,7 +27,7 @@ def as_choices(cls):
         return (
             (cls.DATE, _("Last Seen")),
             (cls.NEW, _("First Seen")),
-            (cls.PRIORITY, _("Priority")),
```
Hm I changed this with the intention of starting to write trends in saved searches instead of priority, but we can do that after the FE is changed.

malwilley replied:
It might be best to have both priority and trends as options here. The frontend decides which options to show ([see here](https://github.com/getsentry/sentry/blob/5fe143b07a16cea81dedc708692b9c48ad0ee26c/static/app/components/modals/savedSearchModal/savedSearchModalContent.tsx#L24-L30)), so I don't think having the extra option here would affect the UI in any way. That way you can modify the frontend to send trends and it won't break (I believe this gets used in the validators when creating a saved search?)

snigdhas committed:
Merge branch 'master' into snigdha/priority-trends-be

snigdhas committed:
Merge branch 'master' into snigdha/priority-trends-be

snigdhas committed:
Add trends back

scttcper committed:
fix(replay): Delay hydration error diff (#65581)

malwilley committed:
feat(suggested-fix): Use new design for processing and results (#65584)

malwilley reviewed (approved):
I think this looks good, might be nice to verify everything works as expected in the UI still

mrduncan committed:
chore(issues): Remove unnecessary wildcards in CODEOWNERS (#65566)

This removes one duplicate rule and converts the other to a directory
match which has the same effect but is easier to validate.

malwilley committed:
feat(issue-priority): Remove issues from stream when reprioritizing (#65551)

Closes https://github.com/getsentry/sentry/issues/65212

When changing the priority to a value that's not shown in the current
query, removes it from the view similar to how it works for making
issues as resolved, archived, etc.

In order to determine whether or not the issue should be removed, parses
the search using `parseIssuePrioritySearch()`. This is a little smarter
than the other issue actions which just check for the query string for
matches like `is:unresolved`.

iamrajjoshi committed:
chore(spike protection): Remove `organizations:spike-protection-decay-heuristic` feature flag (#65488)

Removed `organizations:spike-protection-decay-heuristic` feature flag

Requires: https://github.com/getsentry/getsentry/pull/13010
Fixes:
https://github.com/getsentry/team-core-product-foundations/issues/72

schew2381 committed:
chore: Add logging to find fetched scopes of member (#65579)

Investigating bug in https://github.com/getsentry/sentry/issues/65155

The issue happens because we `self.allowed` roles is not set here
causing the user serialized with authenticators to not be attached to
the response.

https://github.com/getsentry/sentry/blob/eb55184602940adb1fc09a9c0e33515cac92c7d8/src/sentry/api/serializers/models/organization_member/expand/roles.py#L69-L70

We're fetching `allowed_roles` directly in the endpoint logic, which we
then pass to the serializer

https://github.com/getsentry/sentry/blob/9a8c2c709420f95a626bb23b47b3c8097fa1a4c1/src/sentry/api/endpoints/organization_member/details.py#L155-L162

Somewhere in `get_allowed_org_roles`, there's a bug with scopes causing
the function to return an empty list.

ceorourke committed:
ref(weekly reports): Rearrange files & functions to be reused (#65559)

Offshoot of https://github.com/getsentry/sentry/pull/65501 that moves
`weekly_reports.py` into a `summaries` directory and puts classes and
functions to be reused by the daily summary into a utils file.

narsaynorath committed:
fix(app-start): Use consistent colors for releases (#65567)

Use the colors in the mock up for the UI

mrduncan committed:
chore(issues): Update CODEOWNERS to reflect reality (#65580)

This should be the last major pass at changes relating to the issues
team.

JoshFerge committed:
ref(issues): move issues endpoint tests into issues api test folder (#65562)

Commit:
fix(ddm): Fix measurement for metric (#65540)

Commit:
feat(ddm): Allow nested expressions (#65524)

Commit:
fix(dynamic-sampling): Fix performance of the org recalibration cron job (#65539)

Commit:
fix(dynamic-sampling): Fix validation in orgs recalibration (#65604)

Commit:
feat(dynamic-sampling): Return plan sample rate in the organization details (#65605)

ArthurKnaus committed:
feat(ddm): Formulas part one (#65528)

Add button to add formulas to metrics explorer.
Introduce widget types.

- closes https://github.com/getsentry/sentry/issues/64543

TBS1996 committed:
feat(ddm): Implement global abuse limits for metrics by namespace (#65529)

https://github.com/getsentry/relay/issues/2716

Adds the ability to set abuse limits for metric bucket that belong to a
specific namespace.

Commit:
ref(dynamic-sampling): Do not run cron jobs if dynamic sampling is not enabled (#65606)

JonasBa committed:
feat(trace): render web vitals indicators (#65484)

Missing the label rendering, but I'm deferring that just a bit to think
about how I want to handle it.

ArthurKnaus committed:
ref(ddm): Use series id for coloring and identification (#65609)

Assign ids to series to uniquely identify them. Use the id for coloring
and highlighting.

- closes https://github.com/getsentry/sentry/issues/65376

Commit:
feat(dynamic-sampling): Add decorator to recalibrate_orgs_batch (#65610)

shruthilayaj committed:
feat(spans): Register a new experimental group type (#65577)

We are adding support for performance issues with
span streaming. To do this, we're going to hold spans
in a buffer for a few minutes as they arrive, fetch them
from the buffer and run them through the issue detectors.

Registering a new group type to route the issues created
with streamed spans so we can check correct spans/evidence/
fingerprinting as we do dev work.

armenzg committed:
tests(user_misery): Test changes to improve code clarity (#65154)

Changes included:
* Use satisfaction related constants
* Create a testing helper module to discover
* Use `user_misery_formula` in a few more places

cmanallen committed:
fix(replays): Update event-id filter configuration to target legacy error_ids column (#62175)

DominikB2014 committed:
feat(browser-starfish): add use case for bundle analysis (#65267)

markstory committed:
fix(integrations) Pass organization to features.has() (#65569)

In #65433 I neglected to extract the organization from the
RpcOrganizationContext

Fixes SENTRY-2NQR

markstory committed:
fix(hybridcloud) Fix RPC serialization issue for broken integrations (#65578)

When integrations are referenced in alerts, the alert notifier can
survive the orgintegration being disconneced. We shouldn't send `None`
as an id in an RPC call when that happens.

Fixes HC-1106
Fixes SENTRY-2K6J

nhsiehgit committed:
Activated Alert Rule model updates (#65130)

Adds relevant columns for distinguishing between `activated` alert rules
vs. `continuous` alert rules

Surprisingly little is required to get this working 🤔 

We just need to add a monitor_type to the AlertRule itself in order to
determine whether or not to create our subscriptions in Snuba
Once we have that, we only really need to update the SnubaQuery model to
allow us to set a specified `start_time`, then we can get crackin on
actually implementing the start time / dynamic query creation/deletions

cathteng committed:
chore(team-org-roles): remove "get_all_org_roles" from OrganizationMember (#65350)

armenzg committed:
fix(metrics_extraction): `epm` and `eps` to use standard metrics when not applicable (#65612)

The task would call `should_use_on_demand` with `epm()` while the
endpoint would call with `epm(900)`. In either case, we should be using
standard metrics since the query is empty.

wmak committed:
experiment(trace): Add an exit-early parameter (#65619)

- This is to help debug why this endpoint is timing out in prod and only
in prod

Zylphrex committed:
feat(metrics): Support custom samples (#65582)

This adds support for finding span samples for custom metrics.

lobsterkatie committed:
ref(grouping): Small changes to prep for `create_and_seek_grouphashes` introduction (#65269)

This is a bunch of dumb rearrangement pulled out of an upcoming PR in order to make that PR easier to review. No behavior changes, just a bunch of moving stuff around.

Abdkhan14 committed:
feat(trace): Added multiple bars for sibling autogrouping. (#65553)

Rendering multiple bars for sibling autogroup node, depending on the
gaps between its children.

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

cathteng committed:
ref(slack): replace rich text formatted block with quote block in markdown (#65588)

malwilley committed:
feat(issue-priority): Add inbox tab for new search query (#65594)

The default query was already modified to include the new
`issue.priority`, this adds a tab for it.

JonasBa committed:
fix(profiling): skip generating missing spans for browser sdks (#65623)

Mimic the logic from waterfall model and skip generating the missing
spans for browser trees

wedamija committed:
chore(teams): Remove unused `Team.transfer_to` function and tests (#65599)

This code is unused and will not work well if it's ever called. Just
killing it to simplify the codebase

asottile-sentry committed:
ref: use aware datetimes for test_organization_releases (#65626)

this fixes an (ignored) django warning:

```
E   RuntimeWarning: DateTimeField Release.date_added received a naive datetime (2013-08-13 03:08:24.880386) while time zone support is active.
```

<!-- Describe your PR here. -->

ceorourke committed:
ref(slack): Handle rule not existing (#65585)

Fixes SENTRY-2KVF - https://sentry.sentry.io/issues/4958372567/

asottile-sentry committed:
ref: fix tz unaware warning in apidocs test_deploys (#65627)

this warning is currently disabled but causes a lot of spam in
production and will eventually be an error

```
E   RuntimeWarning: DateTimeField Deploy.date_finished received a naive datetime (2024-02-21 17:34:16.042242) while time zone support is active.
```

<!-- Describe your PR here. -->

markstory committed:
fix(devservices) Don't print during module setup (#65614)

We shouldn't echo any content when this module is imported. Instead only
print when commands are run.

schew2381 committed:
chore(staff): Create member and staff permission class (#65598)

Also moved the existing `MemberPermission` class into the base file, and
also created tests for it

lobsterkatie committed:
ref(grouping): Split primary from secondary grouping and add helper (#65273)

In `_save_aggregate_new`, we currently calculate primary and secondary hashes separately, but we then lump them together for the purposes of creating grouphashes and then searching for one associated with a group.

This PR breaks those latter steps apart, such that each stage is done separately for primary and secondary hashes. Then, since that creates two identical processes, it pulls that logic out into a new helper function, `create_and_seek_grouphashes`, which handles all three tasks - calculation, creation, and searching. In order to not just return an enormous tuple from that function, it also creates a dataclass to contain the values it returns.

Note: In order to make the process between primary and secondary hashing truly identical, I had to type `GroupHashInfo` such that all values could be null, because in the case of secondary grouping which doesn't run, they are. This then means that even in the case of primary grouping (where we know for sure we're at least going to get back a grouping config and some hash values), we have to deal with the possibility of Nones - specifically, in this case, by casting them before using them to record metrics. It's not great, but it's fixed in https://github.com/getsentry/sentry/pull/65372.

lobsterkatie committed:
feat(tests): Allow `capture_return_values` to capture thrown errors (#65575)

This modifies our test helper, `capture_return_values`, so that it can also capture results when the function it's wrapping throws an error. This is helpful when spying on a function whose thrown errors are caught, and therefore unable to be examined and asserted about using `pytest.raises`.

Since recording results now happens twice, it's been pulled out into its own helper, `record_result`. Also, since the function now captures more than just return values, this also renames it to `capture_results`.

Commit:
fix(crons): Show all env options when editing monitor notifications (#65592)

<img width="925" alt="image"
src="https://github.com/getsentry/sentry/assets/9372512/587b1b6d-7f99-431c-89ad-62604514cc96">

Show all environment options for this dropdown

Fixes: https://github.com/getsentry/sentry/issues/65442

scttcper committed:
ref(ui): Change renderContextMenu return type (#65601)

scttcper committed:
ref(ui): Change EarlyFeatures form api types (#65631)

snigdhas commented:
Worked locally with and without the FE counterpart
